workorders.lic

```diff
@@ -207,32 +197,27 @@ class WorkOrders
     end
   end
 
-  def sew_items(info, item, quantity)
-    recipe = @recipes.find { |r| r['name'] == item }
+  def order_yarn(stock_room, stock_needed, stock_number)
+    return unless stock_needed > 0
 
-    if 'What were' == bput('get yarn', 'What were', 'You get')
-      walk_to info['stock-room']
+    stock_needed.times do |_|
+      order_item(stock_room, stock_number)
+      combine_yarn
+    end
+  end
 
-      stock_needed = (quantity * recipe['volume'] / 100.0).ceil
+  def sew_items(info, item, quantity)
+    recipe = @recipes.find { |r| r['name'] == item }
 
-      stock_needed.times do |_|
-        order_stock(info['stock-number'])
-        combine_yarn
-      end
-    else
-      length = bput('count my yarn', 'You count out \d+ yards').scan(/\d+/).first.to_i
-      stock_needed = ((quantity * recipe['volume'] - length) / 100.0).ceil
-      if stock_needed > 0
-        walk_to info['stock-room']
-        stock_needed.times do |_|
-          order_stock(info['stock-number'])
-          combine_yarn
-        end
-      end
-    end
+    existing = if 'What were' == bput('get yarn', 'What were', 'You get')
+                 0
+               else
+                 bput('count my yarn', 'You count out \d+ yards').scan(/\d+/).first.to_i
+             end
```

Comment:
Rubocop seems to have a strong opinion on that one
On Jun 2, 2016 8:47 PM, "Robert Uhl" <notifications@github.com> wrote:

> In workorders.lic
> <https://github.com/rpherbig/dr-scripts/pull/417#discussion_r65642926>:
>
> > -    else
> > -      length = bput('count my yarn', 'You count out \d+ yards').scan(/\d+/).first.to_i
> > -      stock_needed = ((quantity * recipe['volume'] - length) / 100.0).ceil
> > -      if stock_needed > 0
> > -        walk_to info['stock-room']
> > -        stock_needed.times do |_|
> > -          order_stock(info['stock-number'])
> > -          combine_yarn
> > -        end
> > -      end
> > -    end
> > +    existing = if 'What were' == bput('get yarn', 'What were', 'You get')
> > +                 0
> > +               else
> > +                 bput('count my yarn', 'You count out \d+ yards').scan(/\d+/).first.to_i
> > +             end
>
> Really not a fan of this formatting, end doesn't seem to line up and it's
> hard to read.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rpherbig/dr-scripts/pull/417/files/ef37bdc24c327dec7104c98824d38f496363a6bd#r65642926>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABIHp-YybnTp7-w3xD1FiuByNuhixhLVks5qH3mlgaJpZM4ItG7x>
> .
>